opm/parser/eclipse/EclipseState/Grid/FieldPropsManager.hpp

```diff
@@ -187,6 +187,59 @@ struct MemInfo {
     virtual bool has_int(const std::string& keyword) const { return this->has<int>(keyword); }
     virtual bool has_double(const std::string& keyword) const { return this->has<double>(keyword); }
 
+    /*
+      The transmissibility keywords TRANX, TRANY and TRANZ do not really fit
+      well in the FieldProps system. The opm codebase is based on a full
+      internalization in the parse phase, and then passing fully assembled
+      objects to the simulator. When it comes to the transmissibilities this
+      model breaks down because the input code in opm-common is not capable of
+      calculating the transmissibility, that is performed in the simulator.
+
+      The EDIT section can have modifiers on TRAN, these must be applied *after*
+      the initial transmissibilities are calculated. To support this all the
+      modifiers to the TRAN{XYZ} fields are assembled in "transmissibility
+      calculators", and then these modifiers can be applied to a TRAN vector
+      after it has been calculated in the simulator. Usage from the simulator
+      could look like:
+
+
+          const auto& fp = eclState.fieldProps();
+
+          // Calculate transmissibilities using grid and permeability
+          std::vector<double> tranx = ....
+
+          // Check if there are any active TRANX modifiers and apply them
+          if (fp.tran_active("TRANX"))
+               fp.apply_tran("TRANX", tranx);
+
+
+    */
+
+    /*
+      Will check if there are any TRAN{XYZ} modifiers active in the deck.
+    */
+    virtual bool tran_active(const std::string& keyword) const;
```

Comment:
any reason why these are virtual ?

Reply:
Ehhh no - the driver does not have a proper C++ license.